builders/interact/verify-contracts/etherscan-plugins.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Etherscan Plugins
+description: Learn about how to verify smart contracts on Moonbeam networks using the Etherscan plugins made available by Hardhat and Truffle
+---
+
+# Verify Smart Contracts with Etherscan Plugins
+
+![Etherscan Plugins Banner](/images/builders/interact/verify-contracts/etherscan-plugins/plugins-banner.png)
+
+## Introduction {: #introduction } 
+
+Verifying smart contracts is a great way of improving the transparency and security of contracts deployed on Moonbeam. There are a couple of plugins that integrate with Etherscan's contract verification service, including the [`hardhat-etherscan` plugin](https://hardhat.org/plugins/nomiclabs-hardhat-etherscan.html){target=blank} and the [`truffle-plugin-verify` plugin](https://github.com/rkalis/truffle-plugin-verify){target=blank}. Both plugins can be used to automate the process of verifying contracts by locally detecting which contracts to verify and which Solidity libraries are required, if any.
+
+The Hardhat plugin integrates seamlessly into your Hardhat project, and likewise the Truffle plugin integrates into your Truffle project.
+
+This guide will show you how to use both plugins to verify smart contracts deployed on Moonbase Alpha. This guide can also be adapted for Moonbeam and Moonriver.
+
+## Checking Prerequisites {: #checking-prerequisites }
+
+To follow along with this tutorial, you will need to have:
+
+- [MetaMask installed and connected to the Moonbase Alpha](/tokens/connect/metamask/){target=blank} TestNet
+- An account funded with `DEV` tokens. You can obtain tokens for testing purposes from the Moonbase Alpha [faucet](/builders/get-started/moonbase/#get-tokens/){target=blank}
+- A [Moonscan API key](#generating-a-moonscan-api-key)
+
+## Generating a Moonscan API Key {: generating-a-moonscan-api-key }
```

Comment:
Careful here. Moonscan/Moonbeam API Key works for Moonbeam and Moonbase Alpha, but you need to go to Moonscan/Moonriver to get an API key that works there